File: test_scripts/r/test1/data_prep.R
# Data preparation - depends on utils.R

source("utils.R")

# Create sample data
set.seed(123)
raw_data <- data.frame(
  id = 1:100,
  value = c(rnorm(80, 50, 10), rep(NA, 20)),  # 20% missing values
  category = sample(c("A", "B", "C"), 100, replace = TRUE)
)

# Clean the data using utility function
clean_dataset <- clean_data(raw_data)

print(paste("Data prepared:", nrow(clean_dataset), "clean records"))

File: test_scripts/python/test1/data_prep.py
"""Data preparation script - depends on utils.py"""

import utils

# Create sample data
raw_data = [10, 20, 30, None, 40, 50, None, 60, 70, 80]

# Clean the data using utility function
clean_dataset = utils.clean_data(raw_data)

# Calculate statistics
stats = utils.calculate_stats(clean_dataset)

print(f"Data prepared: {len(clean_dataset)} clean records")
print(f"Statistics: {stats}")